repair/repair.cc

```diff
@@ -1390,6 +1404,9 @@ future<> repair::data_sync_repair_task_impl::run() {
                         id, std::move(data_centers), std::move(hosts), std::move(ignore_nodes), reason, hints_batchlog_flushed, ranges_parallelism);
                 task_impl_ptr->neighbors = std::move(neighbors);
                 auto task = co_await local_repair._repair_module->make_task(std::move(task_impl_ptr), parent_data);
+                auto release_task_resources = defer([&] () noexcept {
+                    task->release_resources();
+                });
```

Comment:
Can the task manager automatically call this when the task is done?

Reply:
I was thinking about it and did it like this for one reason:
`release_resources` method is useful mainly for tasks which replace already existing classes (like `shard_repair_task_impl` replaced `repair_info`) and thus the lifetime of their object is prolonged.

For both `shard_repair_task_impl` and `compaction_task_executor`s the original object was destroyed immediately after the operation is finished. But this doesn't need to be the case for all future tasks. Let's say an original object isn't destroyed after operation is done, but a shared_ptr to it is stored in some vector and its destruction is invoked by some other event. It would be nice if we could reuse `release_resources` method also in this case.

I'm not sure how probable this situation is and whether we should consider it at this moment. If you don't think that's a valid concern I will change that.

Reply:
I don't have a strong opinion on this. We can change this after we have more tasks and have a perspective on it.